File: ccif/src/java/other/org/himalay/ccif/HistoryCnf.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:57 EST 2016")

public  class HistoryCnf extends   APDUBaseFactory.APDUBase { //Concrete type is HistoryCnf




    // members variables
    // header
    public APDUHeader header ;
    // length
    public short length ;
    // statusField
    public short statusField ;
    // commentLength
    public ByteArray commentLength ;
    // ippvslotNb
    public ByteArray ippvslotNb ;

        public HistoryCnf () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize header
        header = new APDUHeader();
        // Initialize length
        
        // Initialize statusField
        
        // Initialize commentLength
        commentLength= new ByteArray();commentLength.setSizeType("FIRST_UI8");
        // Initialize ippvslotNb
        ippvslotNb= new ByteArray();ippvslotNb.setSizeType("FIRST_UI8");
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
  preRead();
    int retVal= 0;
                // read length
        {length=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read statusField
        {statusField=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read commentLength
        {retVal+=commentLength.read(istream); }
        // read ippvslotNb
        {retVal+=ippvslotNb.read(istream); }

postRead();
        return retVal;
        }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read header
        retVal += header.read(istream);
        // read length
        {length=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read statusField
        {statusField=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read commentLength
        {retVal+=commentLength.read(istream); }
        // read ippvslotNb
        {retVal+=ippvslotNb.read(istream); }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        { /** fix dependent sizes for header **/  }
        
        
        { /** fix dependent sizes for commentLength**/  }
        { /** fix dependent sizes for ippvslotNb**/  }
    
                // write header
        if (header!=null)retVal +=header.write(ostream);
        // write length
        ostream.writeByte(length); retVal +=1;
        // write statusField
        ostream.writeByte(statusField); retVal +=1;
        // write commentLength
        {retVal += commentLength.write(ostream);}
        // write ippvslotNb
        {retVal += ippvslotNb.write(ostream);}
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("HistoryCnf\n");
    dc.increaseIndent();
        int retVal= 0;
        // write header
        if ( header != null ) {dc.indent();dc.getPs().println("header") ;retVal +=header.dump(dc);}
        // write length
        dc.indent();dc.getPs().println("length="+length+"(0x"+ Integer.toHexString(length) + ")" );
        // write statusField
        dc.indent();dc.getPs().println("statusField="+statusField+"(0x"+ Integer.toHexString(statusField) + ")" );
        // write commentLength
        dc.indent();dc.getPs().print("commentLength: "+commentLength.getSize()+"(0x"+Integer.toHexString(commentLength.getSize())+")\n");this.commentLength.dump(dc);
        // write ippvslotNb
        dc.indent();dc.getPs().print("ippvslotNb: "+ippvslotNb.getSize()+"(0x"+Integer.toHexString(ippvslotNb.getSize())+")\n");this.ippvslotNb.dump(dc);
dc.decreaseIndent();
        return retVal;
    }


        // Getter for header
    //public APDUHeader getHeader()
    //{
    //    return header ;
    //}

    
    // Setter for header
    //public void setHeader(APDUHeader val)
    //{
    //    this.header= val;
    //}
    // Getter for length
    //public short getLength()
    //{
    //    return length ;
    //}

    
    // Setter for length
    //public void setLength(short val)
    //{
    //    this.length= val;
    //}
    // Getter for statusField
    //public short getStatusField()
    //{
    //    return statusField ;
    //}

    
    // Setter for statusField
    //public void setStatusField(short val)
    //{
    //    this.statusField= val;
    //}
    // Getter for commentLength
    //public ByteArray getCommentLength()
    //{
    //    return commentLength ;
    //}

    
    // Setter for commentLength
    //public void setCommentLength(ByteArray val)
    //{
    //    this.commentLength= val;
    //}
    // Getter for ippvslotNb
    //public ByteArray getIppvslotNb()
    //{
    //    return ippvslotNb ;
    //}

    
    // Setter for ippvslotNb
    //public void setIppvslotNb(ByteArray val)
    //{
    //    this.ippvslotNb= val;
    //}


    public void setCommentLength(byte[] val)
    {
        this.commentLength.setData(val);
    }
    
    
    public void setIppvslotNb(byte[] val)
    {
        this.ippvslotNb.setData(val);
    }
    
    
    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        
     public void setHeader(APDUHeader header)
    {
         this.header= header;   
    }
    
    public APDUHeader getHeader()
    {
        return this.header;
    }

}

// End of code

File: libccif/src/bp/HistoryCnf.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:14 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "HistoryCnf.h"

#ifdef org_himalay_ccif_HistoryCnf__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
HistoryCnf::HistoryCnf() {
    references= 0;


    // header
    org_himalay_ccif_APDUHeader__NEW( header);
    // length
    length= (ui8)0;
    // statusField
    statusField= (ui8)0;
    // commentLength
    org_himalay_msgs_runtime_ByteArray_NEW(commentLength);
    commentLength->setSizeType("FIRST_UI8");
    // ippvslotNb
    org_himalay_msgs_runtime_ByteArray_NEW(ippvslotNb);
    ippvslotNb->setSizeType("FIRST_UI8");
}
// Destructor
HistoryCnf::~HistoryCnf() {
    // header
    org_himalay_ccif_APDUHeader__DELETE( header);
    // length

    // statusField

    // commentLength
    org_himalay_msgs_runtime_ByteArray_DELETE( commentLength);
    // ippvslotNb
    org_himalay_msgs_runtime_ByteArray_DELETE( ippvslotNb);
}

int HistoryCnf::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // statusField
    {
        statusField=(istream.readUI8());
        retVal+=1;
    }
    // commentLength
    {
        retVal+=commentLength->read(istream);
    }
    // ippvslotNb
    {
        retVal+=ippvslotNb->read(istream);
    }
    return retVal;
}

int HistoryCnf::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read header
    retVal += header->read(istream);
    // read length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // read statusField
    {
        statusField=(istream.readUI8());
        retVal+=1;
    }
    // read commentLength
    {
        retVal+=commentLength->read(istream);
    }
    // read ippvslotNb
    {
        retVal+=ippvslotNb->read(istream);
    }

    return retVal;
}

int HistoryCnf::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for header **/
    }


    {   /** fix dependent sizes for commentLength**/
    }
    {   /** fix dependent sizes for ippvslotNb**/
    }

    int retVal= 0;
    // write header
    if ( header != NULL ) {
        retVal +=header->write(ostream);
    }
    // write length
    ostream.writeUI8(length);
    retVal +=1;
    // write statusField
    ostream.writeUI8(statusField);
    retVal +=1;
    // write commentLength
    {
        retVal += commentLength->write(ostream);
    }
    // write ippvslotNb
    {
        retVal += ippvslotNb->write(ostream);
    }

    return retVal;
}

int HistoryCnf::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("HistoryCnf\n");
    dc.increaseIndent();
    int retVal= 0;
    // write header
    if ( header != NULL ) {
        dc.indent();
        dc.getPs().println("header") ;
        retVal +=header->dump(dc);
    }
    // write length
    dc.indent();
    dc.getPs().print("length=");
    dc.getPs().println((long)length);
    // write statusField
    dc.indent();
    dc.getPs().print("statusField=");
    dc.getPs().println((long)statusField);
    // write commentLength
    dc.indent();
    dc.getPs().print("commentLength");
    commentLength->dump(dc);
    // write ippvslotNb
    dc.indent();
    dc.getPs().print("ippvslotNb");
    ippvslotNb->dump(dc);
    dc.decreaseIndent();
    return retVal;
}


// Getter for header
//org_himalay_ccif_APDUHeader__PTR_TYPE HistoryCnf::getHeader()
//{
//return this->header;
//}

// Setter for header
//void HistoryCnf::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE val)
//{
//this->header= val;
//}

// Getter for length
//ui8 HistoryCnf::getLength()
//{
//return this->length;
//}

// Setter for length
//void HistoryCnf::setLength(ui8 val)
//{
//this->length= val;
//}

// Getter for statusField
//ui8 HistoryCnf::getStatusField()
//{
//return this->statusField;
//}

// Setter for statusField
//void HistoryCnf::setStatusField(ui8 val)
//{
//this->statusField= val;
//}

// Getter for commentLength
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE HistoryCnf::getCommentLength()
//{
//return this->commentLength;
//}

// Setter for commentLength
//void HistoryCnf::setCommentLength(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->commentLength= val;
//}

// Getter for ippvslotNb
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE HistoryCnf::getIppvslotNb()
//{
//return this->ippvslotNb;
//}

// Setter for ippvslotNb
//void HistoryCnf::setIppvslotNb(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->ippvslotNb= val;
//}

int HistoryCnf::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

void HistoryCnf::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE header)
{
    this->header= header;
}

org_himalay_ccif_APDUHeader__PTR_TYPE HistoryCnf::getHeader()
{
    return this->header;
}

#ifdef org_himalay_ccif_HistoryCnf__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::HistoryCnf* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::HistoryCnf* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code